sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -184,6 +184,17 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
       2, 3, 4)
   }
 
+  test("SPARK-16853: select, case class and tuple") {
```

Comment:
@cloud-fan 

This follows existing test case name, for example, for select 2, current name is
```
test("select 2, primitive and tuple") {
```

https://github.com/apache/spark/pull/14474/files#diff-3836bd1fc5ae9c8b8ac4bdb2b9944159L196